File: src/common/src/models/payment/serializedPayment.ts
import type { Payment } from './payment';

export type SerializedPayment = {
  /**
   * id
   */
  i: string;
  /**
   * amount
   */
  a: string;
  /**
   * created
   */
  c: number;
  /**
   * data
   */
  d?: Payment['data'];
  /**
   * asset
   */
  as?: SerializedPaymentAsset;
  /**
   * successUrl
   */
  su?: string;
  /**
   * cancelUrl
   */
  cu?: string;
  /**
   * expired
   */
  e?: number;
  /**
   * signature
   */
  s: SerializedPaymentSignature;
};

export interface SerializedPaymentAsset {
  /**
   * address
   */
  a: string;
  /**
   * decimals
   */
  d: number;
  /**
   * id
   */
  i?: number;
}

export interface SerializedPaymentSignature {
  /**
   * contract
   */
  c: string;
  /**
   * client
   */
  cl?: string;
  /**
   * signingPublicKey
   */
  k: string;
}

export type NonSerializedPaymentSlice = Pick<Payment, 'targetAddress'>;

File: packages/dotnet/TezosPayments/Serialization/SerializedPayment.cs
using System.Text.Json.Serialization;

namespace TezosPayments.Serialization;

public class SerializedPayment
{
    [JsonPropertyName("i")]
    public string? Id { get; set; }

    [JsonPropertyName("a")]
    public string? Amount { get; set; }

    [JsonPropertyName("t")]
    public string? Target { get; set; }

    [JsonPropertyName("as")]
    public SerializedPaymentAsset? Asset { get; set; }

    [JsonPropertyName("d")]
    public object? Data { get; set; }

    [JsonPropertyName("su")]
    public string? SuccessUrl { get; set; }

    [JsonPropertyName("cu")]
    public string? CancelUrl { get; set; }

    [JsonPropertyName("c")]
    public long Created { get; set; }

    [JsonPropertyName("e")]
    public long? Expired { get; set; }

    [JsonPropertyName("s")]
    public SerializedPaymentSignature? Signature { get; set; }
}
